src/Microsoft.AspNet.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -136,6 +139,11 @@ internal static void AddViewServices(IServiceCollection services)
             // These are stateless so their lifetime isn't really important.
             services.TryAddSingleton<ITempDataDictionaryFactory, TempDataDictionaryFactory>();
             services.TryAddSingleton<SaveTempDataFilter>();
+
+            services.TryAddSingleton(ArrayPool<ViewBufferValue>.Create(MemoryPoolViewBufferScope.SegmentSize, 50));
```

Comment:
The default array is pretty crazy long: `private const int DefaultArrayLength = 1048576;` based on reflection. Is the idea that you're supposed to perform sub chunking on the one array that is pooled?